### DISCUSSION TITLE
Working JS (without TS) example on jsfiddle or codepen etc

### DISCUSSION THREAD

**Discussion by zehawki** [react-page/react-page#894] [Q&A] [answered]
Is there a fully working example on jsfiddle, codepen or similar to actually see and experiment with? I am unable to get the "simple example" working: https://react-page.github.io/docs/#/quick-start?id=simple-example

It also quite unclear what < PageLayout > is there for and what does it have to do with the Editor.

**Comment by macrozone** [comment, marked as answer]
@zehawki i tried to host all examples on jsfiddle or codepen, but i had some problems with dynamic imports, so i currently settled on some examples that run on nextjs in the project. This had the additional benefit of showing that server side rendering works properly as well.

I additionaly added an example of using it in "react admin", an admin framework (https://react-page.github.io/examples/reactadmin), because that is how I used it in some real-world projects of mine

So an easy way of trying it out is to clone this repo and running `yarn install` and then `yarn dev`

this will build and watch everything and open a server on localhost:3000 with the examples.

The one "simple example" you mentioned is also one that runs on the examples:

- demo https://react-page.github.io/examples/simple
- source https://github.com/react-page/react-page/blob/master/examples/pages/examples/simple.tsx (its exactly the same as mentioned in https://react-page.github.io/docs/#/quick-start?id=simple-example)

You get a JS version if you just omit all type definitions (there is only one on [line 22](https://github.com/react-page/react-page/blob/master/examples/pages/examples/simple.tsx#L22), but i highly recommend to consider using typescript in general.

It would also be helpful to know what the problems are that you are facing, so that we can try to improve the documentation accordingly.

**Reply by zehawki** [reply]
> You get a JS version if you just omit all type definitions (there is only one on line 22, but i highly recommend to consider using typescript in general.

Yes I did that, that was simple. Then I hacked and slashed my way through the demo - remove various nodes until the bare editor is left. Thats here:

![image](https://user-images.githubusercontent.com/8095144/108202647-738a6b00-7147-11eb-989e-7a3994340b0d.png)


Then I tried to mirror the same div structure with its CSS as follows:

```
    <div style={{ display: "flex" }}>
      <div style={{ flexGrow: 1, padding: 32, backgroundColor: "#eee" }}>
        <div style={{ padding: 32, margin: "auto", backgroundColor: "white" }}>
          <Editor cellPlugins={cellPlugins} value={value} onChange={setValue} />
        </div>
      </div>
    </div>
```

But this is not going to get me far. I can see lot of CSS on the demo page that is not getting applied on my side. Here's my code so far. As you can see its absolutely bare bones.

```
import React, { useState } from 'react';

// The editor core
import Editor, { Value } from '@react-page/editor';

// The rich text area plugin
import slate from '@react-page/plugins-slate';
import '@react-page/plugins-slate/lib/index.css';


// image
// import image from '@react-page/plugins-image';
// import PageLayout from '../../components/PageLayout';
// Stylesheets for the rich text area plugin
// uncomment this


// Stylesheets for the imagea plugin
//import '@react-page/plugins-image/lib/index.css';

// Define which plugins we want to use.
const cellPlugins = [slate()];

export default function SimpleExample() {
  const [value, setValue] = useState(null);

  return (
    <div style={{ display: "flex" }}>
      <div style={{ flexGrow: 1, padding: 32, backgroundColor: "#eee" }}>
        <div style={{ padding: 32, margin: "auto", backgroundColor: "white" }}>
          <Editor cellPlugins={cellPlugins} value={value} onChange={setValue} />
        </div>
      </div>
    </div>

  );
}
```

**Reply by zehawki** [reply]
Ok so I cloned and ran, the editor shows up fine, so thats great, but it doesnt help particularly. Internally its a mess - CSS is applied from various places, eg class _react-page-row_ is in 2 files:

1. /packages/editor/lib/index.css
2. /packages/editor/src/core/grid.css 

On top of that, styles are directly applied in the code to this same element which is using this CSS class.

`<div class="react-page-row" style="position: relative; border-color: red;">`

And none of this CSS is applied when using the component as a package, so when instantiating this as a component it doesnt show things as expected. When I manually add the CSS one by one, things in the editor begin to show up, but how many CSS can one bring in like this? And this is anyway an incorrect way to proceed.

**Example:**

Class _react-page-editable_ is applied from 2 files, but at least it there in the dev environment.

![image](https://user-images.githubusercontent.com/8095144/108233613-fb816c80-7169-11eb-890a-8334904407f6.png)

When using as a packaged component, the CSS is not even there.

![image](https://user-images.githubusercontent.com/8095144/108233947-47341600-716a-11eb-918c-2382c47f57c7.png)

Its impossible to muck around with the dev side of this - if it can be used seamlessly as a packaged component, I'd love to adopt it.

**Reply by macrozone** [reply]
I would also love to have no css and instead do everything with props, but its currently a compromise and overriding styles is easier and simpler with css.

You have to import one CSS:

`import '@react-page/editor/lib/index.css';` 

and if you are using the rich-text plugin (you probably will)

import '@react-page/plugins-slate/lib/index.css';

unfortunatly there is an error on the docs about that and i'll fix it right away

**Reply by zehawki** [reply]
I see, yes so this was the missing import in the doc.

**Reply by zehawki** [reply]
Please update in this one as well: https://react-page.github.io/docs/#/editor?id=intro
